mender-dist-packages-building/README.md

```diff
@@ -0,0 +1,157 @@
+Thread model: posix
+gcc version 8.3.0 (Raspbian 8.3.0-6+rpi1)
+COLLECT_GCC_OPTIONS='-E' '-v'  '-mfloat-abi=hard' '-mfpu=vfp' '-mtls-dialect=gnu' '-marm' '-march=armv6+fp'
+ /usr/lib/gcc/arm-linux-gnueabihf/8/cc1plus -E -quiet -v -imultilib . -imultiarch arm-linux-gnueabihf -D_GNU_SOURCE - -mfloat-abi=hard -mfpu=vfp -mtls-dialect=gnu -marm -march=armv6+fp
+ignoring duplicate directory "/usr/include/arm-linux-gnueabihf/c++/8"
+ignoring nonexistent directory "/usr/local/include/arm-linux-gnueabihf"
+ignoring nonexistent directory "/usr/lib/gcc/arm-linux-gnueabihf/8/../../../../arm-linux-gnueabihf/include"
+#include "..." search starts here:
+#include <...> search starts here:
+ /usr/include/c++/8
+ /usr/include/arm-linux-gnueabihf/c++/8
+ /usr/include/c++/8/backward
+ /usr/lib/gcc/arm-linux-gnueabihf/8/include
+ /usr/local/include
+ /usr/lib/gcc/arm-linux-gnueabihf/8/include-fixed
+ /usr/include/arm-linux-gnueabihf
+ /usr/include
+End of search list.
+# 1 "<stdin>"
+# 1 "<built-in>"
+# 1 "<command-line>"
+# 1 "/usr/include/stdc-predef.h" 1 3 4
+# 1 "<command-line>" 2
+# 1 "<stdin>"
+COMPILER_PATH=/usr/lib/gcc/arm-linux-gnueabihf/8/:/usr/lib/gcc/arm-linux-gnueabihf/8/:/usr/lib/gcc/arm-linux-gnueabihf/:/usr/lib/gcc/arm-linux-gnueabihf/8/:/usr/lib/gcc/arm-linux-gnueabihf/
+LIBRARY_PATH=/usr/lib/gcc/arm-linux-gnueabihf/8/:/usr/lib/gcc/arm-linux-gnueabihf/8/../../../arm-linux-gnueabihf/:/usr/lib/gcc/arm-linux-gnueabihf/8/../../../:/lib/arm-linux-gnueabihf/:/lib/:/usr/lib/arm-linux-gnueabihf/:/usr/lib/
+COLLECT_GCC_OPTIONS='-E' '-v'  '-mfloat-abi=hard' '-mfpu=vfp' '-mtls-dialect=gnu' '-marm' '-march=armv6+fp'
+```
+
+And gives us a few key data-points, like the sysroot the compiler is configured
+to use. This is the place in which we want to install our packages, in order to
+have the compiler and linker pick them up automatically, without us having to
+juggle `-isystem, -I, -L -l, -rpath, -rpath-link` in an infinite dance.
+
+Also note that the compiler is configured with: `-march=armv6`. Which is what is
+actually missing in the upstream debian pre-packaged cross-compilers.
+
+# How to figure out the versions of libc on a given platform
+
+```bash
+
+GCC_FEATURES=$(gcc -dM -E - <<< "#include <features.h>")
+
+if grep -q __UCLIBC__ <<< "${GCC_FEATURES}"; then
+    echo "uClibc"
+    grep "#define __UCLIBC_MAJOR__" <<< "${GCC_FEATURES}"
+    grep "#define __UCLIBC_MINOR__" <<< "${GCC_FEATURES}"
+    grep "#define __UCLIBC_SUBLEVEL__" <<< "${GCC_FEATURES}"
+elif grep -q __GLIBC__ <<< "${GCC_FEATURES}"; then
+    echo "glibc"
+    grep "#define __GLIBC__" <<< "${GCC_FEATURES}"
+    grep "#define __GLIBC_MINOR__" <<< "${GCC_FEATURES}"
+else
+    echo "something else"
+fi
+```
+
+Which will give you something like:
+
+```
+#define __GLIBC__ 2
+#define __GLIBC_MINOR__ 28
+```
+
+Also check the libstcxx version:
+
+```bash
+glibccpberrypi:~$ strings /usr/lib/arm-linux-gnueabihf/libstdc++.so.6 | grep -i
```

Comment:
grep -i _what_?

Reply:
Shite, fixing in separate commit, merging now

Reply:
supposed to be glibc